### ISSUE TITLE
Incompatibility with dplyr with factor

### ISSUE THREAD

**Issue by DanChaltiel** [gdemin/expss#65]
Hi Gregory,

This one is a bit tricky and you might not want to dig so deep. Still, as I am probably not the only one using `expss` and `dplyr` alongside, I thought you would want to know about it anyway.

When using factors in a `mutate` or `transmute` call, unlike in base `R`, setting the label turns the variable to an integer. The label is rightly set though.

Here is a reproducible example:

```r

library(expss, warn.conflicts = FALSE)
#> 
#> Use 'expss_output_rnotebook()' to display tables inside R Notebooks.
#>  To return to the console output, use 'expss_output_default()'.

#base R
df = mtcars["vs"]
df$vs2 = factor(df$vs, levels=0:1, labels=c("straight", "vshaped"))
df$vs3 = set_var_lab(df$vs2, "Engine shape")

head(df)
#>                   vs      vs2      vs3
#> Mazda RX4          0 straight straight
#> Mazda RX4 Wag      0 straight straight
#> Datsun 710         1  vshaped  vshaped
#> Hornet 4 Drive     1  vshaped  vshaped
#> Hornet Sportabout  0 straight straight
#> Valiant            1  vshaped  vshaped
var_lab(df$vs3)
#> [1] "Engine shape"
class(df$vs3)
#> [1] "labelled" "factor"

#dplyr
library(dplyr, warn.conflicts = FALSE)
df2 = mtcars["vs"] %>% mutate(
    vs2 = factor(df$vs, levels=0:1, labels=c("straight", "vshaped")),
    vs3 = set_var_lab(vs2, "Engine shape")
)

head(df2)
#>   vs      vs2 vs3
#> 1  0 straight   1
#> 2  0 straight   1
#> 3  1  vshaped   2
#> 4  1  vshaped   2
#> 5  0 straight   1
#> 6  1  vshaped   2
var_lab(df2$vs3)
#> [1] "Engine shape"
class(df2$vs3)
#> [1] "labelled" "integer"

#Created on 2020-07-10 by the reprex package (v0.3.0)
```

**Comment by gdemin** [comment]
Hi Dan,

It seems `mutate` calls `c` internally for labelled factors (`vs3`) but not for factors `vs2`. You can easily check it with `debug(c)`. Debugging will called only once.
And  `base::c` treats factors (even unlabelled) badly:
```R
str(c(df$vs2, df$vs2))
# int [1:64] 1 1 2 2 1 2 1 2 2 2 ...
```
I will fix this behavior for labelled factors in the next version.

**Comment by gdemin** [comment]
Fixed in v0.10.6
